internal/engine/upper.go

```diff
@@ -881,3 +883,28 @@ func handleMetricsModeAction(state *store.EngineState, action metrics.MetricsMod
 func handleMetricsDashboardAction(state *store.EngineState, action metrics.MetricsDashboardAction) {
 	state.MetricsServing.GrafanaHost = action.GrafanaHost
 }
+
+func handleOverrideTriggerModeAction(ctx context.Context, state *store.EngineState,
+	action server.OverrideTriggerModeAction) {
+	// TODO(maia): in this implementation, overrides do NOT persist across Tiltfile loads
+	//   (i.e. the next Tiltfile load will wipe out the override we just put in place).
+	//   If we want to keep this functionality, the next step is to store the set of overrides
+	//   on the engine state, and whenever we load the manifest from the Tiltfile, apply
+	//   any necessary overrides.
+
+	// We validate trigger mode when we receive a request, so this should never happen
+	if !model.ValidTriggerMode(action.TriggerMode) {
+		logger.Get(ctx).Errorf("Error overriding trigger mode: invalid trigger mode %d", action.TriggerMode)
```

Comment:
I think it's good policy that "should never happen" error messages explicitly include "internal error" so that if a user *does* hit it, they can skip debugging and go straight to reporting.